stream_alert/alert_processor/outputs.py

```diff
@@ -803,3 +805,296 @@ def dispatch(self, **kwargs):
+        """Establish a cookie based Jira session via basic user auth.
+
+        Args:
+            username (str): The Jira username used for establishing the session
+            password (str): The Jira password used for establishing the session
+
+        Returns:
+            str: Header value intended to be passed with every subsequent Jira request
+        """
+        login_url = os.path.join(self.base_url, self.LOGIN_ENDPOINT)
+        auth_info = {'username': username, 'password': password}
+        header_value = ''
+
+        resp = self._post_request(login_url,
+                                  data=auth_info,
+                                  headers=self._get_default_headers(),
+                                  verify=False)
+
+        try:
+            resp_dict = resp.json()
+            header_value = '{}={}'.format(resp_dict['session']['name'],
+                                          resp_dict['session']['value'])
+        except (ValueError, KeyError):
+            LOGGER.error('An error occurred while establishing Jira session [%s] : [%s]',
+                         resp.status_code,
+                         resp.content)
+
+        return header_value
+
+    def dispatch(self, **kwargs):
+        """Send alert to Jira
+
+        Args:
+            **kwargs: consists of any combination of the following items:
+                descriptor (str): Service descriptor (ie: slack channel, pd integration)
+                rule_name (str): Name of the triggered rule
+                alert (dict): Alert relevant to the triggered rule
+        """
+        creds = self._load_creds(kwargs['descriptor'])
+        if not creds:
+            return self._log_status(False)
+
+        jira_id = None
+        comment_id = None
+        success = False
+        issue_summary = 'StreamAlert {}'.format(kwargs['rule_name'])
+        alert_body = '{{code:JSON}}{}{{code}}'.format(json.dumps(kwargs['alert']))
+        self.base_url = creds['url']
+        self.auth_cookie = self._establish_session(creds['username'], creds['password'])
+
+        # Validate successful authentication
+        if not self.auth_cookie:
+            return self._log_status(success)
+
+        # Attempt to add alert to an existing issue.
+        if creds.get('aggregate', '').lower() == 'yes':
+            jira_id = self._get_existing_issue(issue_summary, creds['project_key'])
+            if jira_id:
+                comment_id = self._create_comment(jira_id, alert_body)
+
+        # Create new issue if aggregation is disabled or existing issue not found.
+        if not jira_id:
+            jira_id = self._create_issue(issue_summary,
+                                         creds['project_key'],
+                                         creds['issue_type'],
+                                         alert_body)
+
+        if jira_id or comment_id:
+            success = True
+            LOGGER.debug('sending alert to Jira %s', jira_id)
```

Comment:
This logger wouldn't tell us if there was a comment added, only if an issue was created right? I'd suggest moving  the logger calls to closer to where the action happens (ie: after `comment_id = self....` above and add two others after the `_get_existing_issue` and `_create_issue` call. It'll help us track the flow in cloudwatch logs as well.

Then you can simply return this at the end:
`return self._log_status(jira_id or comment_id)`

Since `_log_status` will log and error similar to what you're logging here